packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -55,14 +52,15 @@ def download(self, format, mimetype):
         # content, and then virtually clicks it. Kind of magical, but it
         # works...
         element = document.createElement('a')
-        data = io.BytesIO()
-        try:
-            self.canvas.figure.savefig(data, format=format)
-        except Exception:
-            raise
-        element.setAttribute('href', 'data:{};base64,{}'.format(
-            mimetype, base64.b64encode(data.getvalue()).decode('ascii')))
+
+        canvas = self.canvas.get_element('canvas')
+        imgURL = canvas.toDataURL('image/png')
+        element.setAttribute('href', imgURL)
         element.setAttribute('download', 'plot.{}'.format(format))
+        element.dataset.downloadurl = """image/png:{0}:
+                                         {1}""".format(element.download,
```

Comment:
Is the newline after the `{0}:` required?

Reply:
Yes, `flake8` says the line is over 79 characters otherwise.

Reply:
Ok -- the problem with the solution, then is that it inserts an unnecessary newline into the string (which might cause problems on some browsers).

Maybe do something like:

```
element.dataset.downloadurl = (
    "image/png:{0}:{1}".format(
```